client/js/shout.js

```diff
@@ -224,31 +224,59 @@ $(function() {
 				}
 			});
 
-		if (!chan.hasClass("channel")) {
-			return;
-		}
-
-		var type = data.msg.type;
-		if (type === "message" || type === "action") {
-			var nicks = chan.find(".users").data("nicks");
-			if (nicks) {
-				var find = nicks.indexOf(from);
-				if (find !== -1 && typeof move === "function") {
-					move(nicks, find, 0);
+		if (chan.hasClass("channel")) {
+			var type = data.msg.type;
+			if (type === "message" || type === "action") {
+				var nicks = chan.find(".users").data("nicks");
+				if (nicks) {
+					var find = nicks.indexOf(from);
+					if (find !== -1 && typeof move === "function") {
+						move(nicks, find, 0);
+					}
 				}
 			}
 		}
+
+		return msg;
+	}
+
+	function renderChannelMessages(data) {
+		var documentFragment = $(document.createDocumentFragment());
+		for (var i = 0; i < data.messages.length; i++) {
+			var msg = renderChatMessage({
+				chan: data.id,
+				msg: data.messages[i]
+			});
+			documentFragment.append(msg);
+		}
```

Comment:
Similarly to @williamboman's comment, this can be nicely achieved in functional style with the `reduce` method:

```js
var documentFragment = data.messages.reduce(function(df, message) {
	df.append(renderChatMessage({
		chan: data.id,
 		msg: message
	}));
	return df;
}, $(document.createDocumentFragment()));
```